model/protonet.py

```diff
@@ -0,0 +1,100 @@
+import torch.nn.functional as F
+import numpy as np
+import torch
+from torch import nn
+
+class InterpolateModule(nn.Module):
+	"""
+	This is a module version of F.interpolate (rip nn.Upsampling).
+	Any arguments you give it just get passed along for the ride.
+	"""
+
+	def __init__(self, *args, **kwdargs):
+		super().__init__()
+
+		self.args = args
+		self.kwdargs = kwdargs
+
+	def forward(self, x):
+		return F.interpolate(x, *self.args, **self.kwdargs)
+
+class InterpolateModule(nn.Module):
+	"""
+	This is a module version of F.interpolate (rip nn.Upsampling).
+	Any arguments you give it just get passed along for the ride.
+	"""
+
+	def __init__(self, *args, **kwdargs):
+		super().__init__()
+
+		self.args = args
+		self.kwdargs = kwdargs
+
+	def forward(self, x):
+		return F.interpolate(x, *self.args, **self.kwdargs)
+	
+
+def make_net(in_channels, conf, include_last_relu=True):
+    """
+    A helper function to take a config setting and turn it into a network.
+    Used by protonet and extrahead. Returns (network, out_channels)
+    """
+    def make_layer(layer_cfg):
+        nonlocal in_channels
+        
+        # Possible patterns:
+        # ( 256, 3, {}) -> conv
+        # ( 256,-2, {}) -> deconv
+        # (None,-2, {}) -> bilinear interpolate
+        # ('cat',[],{}) -> concat the subnetworks in the list
+        #
+        # You know it would have probably been simpler just to adopt a 'c' 'd' 'u' naming scheme.
+        # Whatever, it's too late now.
+        if isinstance(layer_cfg[0], str):
```

Comment:
Instead of these conditional statements, just hard code for our use case. We don't need config files and special cases